pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -287,3 +291,10 @@ def _should_track_batch_outputs_for_epoch_end(self) -> bool:
         if self.trainer.testing:
             return is_overridden("test_epoch_end", model)
         return is_overridden("validation_epoch_end", model)
+
+    def _initialize_batch_idx_tracker(self, num_dataloaders: int) -> None:
+        self._seen_batches = [0] * num_dataloaders
+
+    def _increment_batch_idx_tracker(self, dataloader_idx: int) -> None:
+        if not self.trainer.sanity_checking:
+            self._seen_batches[dataloader_idx] += self.batch_progress.current.completed
```

Comment:
Why not do this one by one inside `advance` where you get the `dl_eval_step`?

Reply:
I thought since we already have a tracker, why not use that? batch_progress resets after each dataloader call.